src/Sentry/Internal/DebugStackTrace.cs

```diff
@@ -497,14 +497,19 @@ private static void DemangleLambdaReturnType(SentryStackFrame frame)
     [UnconditionalSuppressMessage("SingleFile", "IL3002:Avoid calling members marked with 'RequiresAssemblyFilesAttribute' when publishing as a single-file", Justification = AotHelper.SuppressionJustification)]
     private static PEReader? TryReadAssemblyFromDisk(Module module, SentryOptions options, out string? assemblyName)
     {
+#if !NET8_0_OR_GREATER
+#pragma warning disable 0162
```

Comment:
Where do you get these warnings? Also, I'm not a fan of having negative one line ifdefs like this, makes the code quite hard to read.

Reply:
Rider gets all red without this for me, since statically knows that `null` is impossible. Including ( believe, didn't check yet) on ur suggestion above

Reply:
but doesn't it respect `#pragma warning disable CS0162`?

Reply:
looks like it should since 2023.1 
https://youtrack.jetbrains.com/issue/RSRP-490648

Reply:
I've tested this myself and created an alternative PR: #2994